habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java

```diff
@@ -147,6 +193,24 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             log.info("Checking for updates to poetry-monorepo-dependency-plugin...");
             poetryHelper.installPoetryPlugin("poetry-monorepo-dependency-plugin");
         }
+
+        String username = null;
+        String password = null;
+
+        if (StringUtils.isNotEmpty(this.pypiRepoId)) {
+            username = HabushuUtil.findUsernameForServer(this.settings, this.pypiRepoId);
+            password = HabushuUtil.findPasswordForServer(settings, pypiRepoId, decryptPassword);
+        }
+
+        if (StringUtils.isNotEmpty(pypiRepoUrl)) {
```

Comment:
If the pypiRepoUrl is empty, should we log an error or throw and exception?  Not sure what the best approach is here.